From 2e976e0769b630dc52c6d84752b4560b7a1e897a Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Mon, 8 Jul 2019 20:11:38 +0800
Subject: [PATCH 2/9] Revert "plat: imx8mm: change DISPMIX power domain
 handling"

This reverts commit 58fc5c736781d04af26e5d67836472e2b706a620.

(cherry picked from commit fecb1ef2053b951c37897e188a9feda7fe56a7f3)
---
 plat/imx/imx8mm/gpc.c | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/plat/imx/imx8mm/gpc.c b/plat/imx/imx8mm/gpc.c
index 7fc16edc1..09a61021a 100644
--- a/plat/imx/imx8mm/gpc.c
+++ b/plat/imx/imx8mm/gpc.c
@@ -818,11 +818,11 @@ static void imx_gpc_pm_domain_enable(uint32_t domain_id, uint32_t on)
 		}
 
 		if (domain_id == DISPMIX) {
-			/* de-reset bus_blk rstn and
-			 * enable bus_blk clk
-			 */
-			mmio_write_32(0x32e28000, 0x40);
-			mmio_write_32(0x32e28004, 0x1000);
+			/* special setting for DISPMIX */
+			mmio_write_32(0x303845d0, 0x3);
+			mmio_write_32(0x32e28004, 0x1fff);
+			mmio_write_32(0x32e28000, 0x7f);
+			mmio_write_32(0x32e28008, 0x30000);
 		}
 
 		/* handle the ADB400 sync */
@@ -920,11 +920,6 @@ static void imx_gpc_pm_domain_enable(uint32_t domain_id, uint32_t on)
 			while (mmio_read_32(IMX_GPC_BASE + PU_PGC_DN_TRG) & GPU3D_PWR_REQ);
 		}
 
-		if (domain_id == DISPMIX) {
-			mmio_write_32(0x32e28000, 0x0);
-			mmio_write_32(0x32e28004, 0x0);
-		}
-
 		/* HSIOMIX has no PU bit, so skip for it */
 		if (domain_id != HSIOMIX) {
 			/* set the PGC bit */
-- 
2.17.1

